src/common/browser-adapters/webextension-browser-adapter.ts

```diff
@@ -91,12 +91,22 @@ export abstract class WebExtensionBrowserAdapter
         details: ExtensionTypes.InjectDetails,
     ): Promise<any[]> {
         this.verifyPathCompatibility(details.file);
-        return browser.tabs.executeScript(tabId, details);
+        return typeof browser.tabs.executeScript === 'function'
+            ? browser.tabs.executeScript(tabId, details)
+            : chrome.scripting.executeScript({
+                  target: { tabId, allFrames: details.allFrames },
+                  files: [details.file],
+              } as chrome.scripting.ScriptInjection);
```

Comment:
Is the `as chrome.scripting.ScriptInjection` necessary here (and on L109)? I'd have expected the option block we're passing to match the expected type without requiring a cast